content/300-guides/200-deployment/110-deployment-guides/001-deploying-to-aws-lambda.mdx

```diff
@@ -0,0 +1,305 @@
+
+In this guide, you do the following:
+
+- Install Prisma
+- Connect to your database with Prisma
+- Create a Prisma schema file (`schema.prisma`) from your existing database and generate a client
+- Use `prisma migrate` to mark your schema as applied
+
+## 1. Install Prisma
+
+```terminal
+npm install -D prisma && npm install @prisma/client
+```
+
+## 2. Set the DATABASE_URL in a <inlinecode>.env</inlinecode> file
+
+The Lambda functions need the `DATABASE_URL` environment variable to access the database. You must define an `.env` file. You can then use [`serverless-dotenv-plugin`](https://www.serverless.com/plugins/serverless-dotenv-plugin) to inject environment variables into the function runtime.
+
+Open or create `.env` and set the `DATABASE_URL` to your database credentials. The format is as follows:
+
+```file
+DATABASE_URL="postgres://__USER__:__PASSWORD__@__HOST__/__DATABASE__"
+```
+
+Replace the uppercase placeholders with your database credentials. For example:
+
+```file
+DATABASE_URL="postgres://janedoe:randompassword@yourpostgres.supabase.co:6543/yourdbname"
+```
+
+> **Note:** If your connection URL contains any special characters, then you must [escape them](/reference/database-reference/connection-urls#special-characters).
+
+<Admonition type="warning">
+
+It's best practice to keep secret information, such as `DATABASE_URL`, out of your git repository. To do this, add your `.env` file to `.gitignore`.
+
+</Admonition>
+
+## 3. Create a Prisma schema from your database
+
+`prisma db pull` takes the tables and relations that you define in your database and pulls them into a `schema.prisma` file. In the root of your project, run:
+
+```terminal
+npx prisma db pull
+```
+
+This command also generates Prisma Client based on your Prisma schema. After your `schema.prisma` file has been created, be sure to [add the correct `binaryTarget` to your schema.](#binary-targets-in-schema.prisma)
+
+## 4. Baseline your database
+
+After your schema has been created, be sure to [baseline your database](https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/relational-databases/baseline-your-database-typescript-postgresql) in order to let Prisma know that it should not try to migrate anything (yet!)
+
+From your project root (or the directory that contains your `prisma/` directory) run
+
+```terminal
+mkdir -p prisma/migrations/0_init
+npx prisma migrate diff --from-empty --to-schema-datamodel prisma/schema.prisma --script > prisma/migrations/0_init/migration.sql
+```
+
+These commands will create an initial, empty migration file and then populate it with the current structure of your database. After you've confirm that the `0_init` file looks good, mark it as applied with the command:
+
+```terminal
+npx prisma migrate resolve --applied 0_init
+```
+
+### Result
+
+You should now have a `schema.prisma` and Prisma Client that match your existing database schema. Prisma and your database will also be in sync.
```

Comment:
I think we can do without this section. This goes back to the earlier comment on establishing a reasonable baseline for our deployment guide